### PR TITLE
Updates for melee enchantment bonuses

### LINKED ISSUE #48674: Display non-conventional damage and resistances, passive relic effects on items
# Is your feature request related to a problem? Please describe.

Magiclysm, AFS, probably some vanilla items too have damage/resistance against elements such as biological, electric, pure etc.
The Player has no way to know those values without checking in the repo.

Same with various passive stat boosts, speed modifiers, mana regen. Although this might be a problem since some items are supposed to be "unknown", maybe some new Flag?

# Describe the solution you'd like

Display those values if exist.

# Describe alternatives you've considered

Require some skill check (electronic/fabrication, in Magiclysm replaced with spellcraft?) to show them. The problem is we don't distinguish in any way what category those items belong too. So this approach makes little sense.

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
#77662 was merged, so i decided to do a little sanity sweep around it
Close #48674
#### Describe the solution
If possible, bonus for melee is moved from enchantment to be direclty applied to the melee weapon itself
Few stuff got conditions updated, like bombastic perks `Slice n' Dice`, `Bash n' Mash` and `Knifey` now work only when your own melee damage of related type is higher than 1
New flag RELIC_PINK is added, to handle the fact this items are not highlighted in purple without relic_data
UI now shows all damage types item has, not only physical; Vaniila won't be affected because items here do not deal elemental damage, and mods suffer from presenting incorrect information
#### Testing
image
image
#### Additional context
I wonder how mad it would be to implement dbl_or_var for items

### CODE REVIEW

File: data/mods/Magiclysm/Spells/attunements/Crusader.json [hunk 1]
```diff
@@ -120,14 +120,20 @@
...
+      "MAGIC_FOCUS",
+      "TRADER_AVOID",
+      "SHEATH_SWORD",
+      "RELIC_PINK"
+    ],
     "to_hit": { "grip": "weapon", "length": "long", "surface": "line", "balance": "good" },
     "weight": "1067 g",
     "volume": "2750 ml",
     "longest_side": "120cm",
-    "relic_data": {
```
Comment (b3brodie) [inline data/mods/Magiclysm/Spells/attunements/Crusader.json:128]: It looks like pure damage was not added to the damage definitions for the items in this file after it was removed from the relic data

Comment (GuardianDll) [reply data/mods/Magiclysm/Spells/attunements/Crusader.json:128]: Yes, it was intentional decision, they already have light damage bonus defined, defining two damage bonuses is a mistake

**Comment by Night-Pryanik** [comment]
@GuardianDll will this close #48674?

**Comment by GuardianDll** [comment]
Yes

File: data/json/mutations/mutations.json [hunk 2]
```diff
@@ -3804,7 +3804,12 @@
     "points": 1,
     "vitamin_cost": 60,
     "visibility": 1,
-    "enchantments": [ { "melee_damage_bonus": [ { "type": "stab", "add": 0.5 } ] } ],
+    "enchantments": [
+      {
+        "condition": { "and": [ { "not": "u_has_weapon" }, { "math": [ "u_encumbrance('hand_l') + u_encumbrance('hand_r') <= 0" ] } ] },
```
Comment (Salty-Panda) [inline data/json/mutations/mutations.json:3809]: My knowledge of the game might be outdated but there were fingerless gloves which would illogically block this mutation from working as they add encumbrance

Comment (GuardianDll) [reply data/json/mutations/mutations.json:3809]: Imo i would personally removed this enchantment at all, it's too small to be used at all

**Pull request merged**